com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -57,49 +57,94 @@ internal struct AgentAction
-    /// little may have changed between successive steps.
+    /// Create an agent by extending the Agent class. Add your Agent subclass to
+    /// a [GameObject] in the [Unity scene] that serves as the agent's environment.
+    /// 
+    /// Agents in an environment operate in *steps*. The <see cref="Academy"/> object controls
+    /// step progression and each Agent instance tracks its own step count. A step
+    /// corresponds to one Unity [FixedUpdate] cycle. At each step, an agent collects
+    /// observations, passes them to its decision-making policy, and receives an
+    /// action vector in response. (You can make observations at a lower frequency, if desired.)
+    /// 
+    /// Assign a decision making policy to an agent using a <see cref="BehaviorParameters"/>
+    /// component attached to the agent's [GameObject]. The <see cref="BehaviorType"/> setting
+    /// determines how decisions are made:
+    /// 
+    /// * <see cref="BehaviorType.Default"/>: decisions are made by the external process, 
+    ///   when connected. Otherwise, decisions are made using inference. If no inference model
+    ///   is specified in the BehaviorParameters component, then heuristic decision
+    ///   making is used.
+    /// * <see cref="BehaviorType.InferenceOnly"/>: decisions are always made using the trained
+    ///   model specified in the <see cref="BehaviorParameters"/> component.
+    /// * <see cref="BehaviorType.HeuristicOnly"/>: when a decision is needed, the agent's
+    ///   <see cref="Heuristic"/> function is called. Your implementation is responsible for
+    ///   providing the appropriate action.
     ///
-    /// At any step, an agent may be considered done due to a variety of reasons:
-    ///     - The agent reached an end state within its environment.
-    ///     - The agent reached the maximum # of steps (i.e. timed out).
-    ///     - The academy reached the maximum # of steps (forced agent to be done).
+    /// To trigger an agent decision automatically, you can attach a <see cref="DecisionRequester"/>
+    /// component to the Agent game object. You can also call the agent's <see cref="RequestDecision"/>
+    /// function manually. You only need to call <see cref="RequestDecision"/> when the agent is
+    /// in a position to act upon the decision. In many cases, this will be every [FixedUpdate]
+    /// callback, but could be less frequent. For example, an agent that hops around its environment
+    /// can only take an action when it touches the ground, so several frames might elapse between
+    /// one decision and the need for the next.
+    /// 
+    /// Build the vector containing an agent's observations in the <see cref="CollectObservations"/>
+    /// function. The Agent class calls this function before it uses the observation vector
+    /// to make a decision. Note that you can use visual and raycast observations instead of, or in
```

Comment:
Instead of limiting this to "visual and raycast", can we say something like "Sensor-based observations (such as visual and raycast observations)..."?

Reply:
Except VectorSensor is also sensor-based (it implements ISensor), so stating it that way seems confusing.

Reply:
I rewrote the paragraph. Hopefully that addresses your concern here.